src/app/modules/shared-components/components/path-suggestion/path-suggestion.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnChanges, OnDestroy } from '@angular/core';
+import {
+  BreadcrumbsFromRoot,
+  GetBreadcrumbsFromRootsService
+} from '../../../item/http-services/get-breadcrumbs-from-roots.service';
+import { ReplaySubject, Subject, switchMap } from 'rxjs';
+import { mapToFetchState } from '../../../../shared/operators/state';
+import { map } from 'rxjs/operators';
+import { itemRoute } from '../../../../shared/routing/item-route';
+import { ItemRouter } from '../../../../shared/routing/item-router';
+
+@Component({
+  selector: 'alg-path-suggestion',
+  templateUrl: './path-suggestion.component.html',
+  styleUrls: [ './path-suggestion.component.scss' ],
+})
+export class PathSuggestionComponent implements OnDestroy, OnChanges {
+  @Input() itemId?: string;
+
+  private readonly itemId$ = new ReplaySubject<string>(1);
+  private readonly refresh$ = new Subject<void>();
+
+  state$ = this.itemId$.pipe(
+    switchMap(itemId => this.getBreadcrumbsFromRootsService.get(itemId).pipe(
+      map(group => {
+        if (group.length === 0) {
+          return undefined;
+        }
+
+        const allBreadcrumbs = group
+          .map(items => items.slice(0, items.length - 1))
+          .reduce((state, items) => [ ...state, ...items ], []);
```

Comment:
FYI: This is the equivalent as the `[].flat(1)` array function I think. [See MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)

Reply:
App tsconfig.json lib settings is `es2018`, `flat` requires es2019

Reply:
Anyway next fix will remove it